src/server/Services/GatewayService.cs

```diff
@@ -2,22 +2,34 @@
 using System.Reflection;
 using chia.dotnet;
 using System.Web;
+using System.Text.Json;
 
 namespace dig.server;
 
-public sealed class GatewayService(DataLayerProxy dataLayer,
-                                    ChiaConfig chiaConfig,
-                                    StoreRegistryService storeRegistryService,
-                                    IMemoryCache memoryCache,
-                                    ILogger<GatewayService> logger,
-                                    IConfiguration configuration)
+public class GatewayService
 {
-    private readonly DataLayerProxy _dataLayer = dataLayer;
-    private readonly ChiaConfig _chiaConfig = chiaConfig;
-    private readonly StoreRegistryService _storeRegistryService = storeRegistryService;
-    private readonly IMemoryCache _memoryCache = memoryCache;
-    private readonly ILogger<GatewayService> _logger = logger;
-    private readonly IConfiguration _configuration = configuration;
+    private readonly DataLayerProxy _dataLayer;
+    private readonly ChiaConfig _chiaConfig;
+    private readonly StoreRegistryService _storeRegistryService;
+    private readonly IMemoryCache _memoryCache;
+    private readonly ILogger<GatewayService> _logger;
+    private readonly IConfiguration _configuration;
+    private FileCacheService _fileCache;
+    private readonly StoreUpdateNotifierService _storeUpdateNotifierService;
+
+    public GatewayService(DataLayerProxy dataLayer, ChiaConfig chiaConfig, StoreRegistryService storeRegistryService, IMemoryCache memoryCache, ILogger<GatewayService> logger, IConfiguration configuration)
+    {
+        _dataLayer = dataLayer;
+        _chiaConfig = chiaConfig;
+        _storeRegistryService = storeRegistryService;
+        _memoryCache = memoryCache;
+        _logger = logger;
+        _configuration = configuration;
+        _fileCache = new FileCacheService(@"C:\Temp\store-cache", _logger);
```

Comment:
@dkackman please help me get this into a more appropriate folder

Reply:
added what what should work, don't have a chance to test it but that is the gist.